### PR TITLE
Guard against popping item from empty vector (list_elements_stack)

### PR DESCRIPTION
**Pull request by dennisschagt**
Fixes https://github.com/newsboat/newsboat/issues/1974

### CODE REVIEW

File: src/htmlrenderer.cpp [hunk 1]
```diff
@@ -683,7 +683,9 @@ void HtmlRenderer::render(std::istream& input,
 						tables.size() ? 0
 						: indent_level);
 				}
-				list_elements_stack.pop_back();
+				if (!list_elements_stack.empty()) {
+					list_elements_stack.pop_back();
+				}
```
Comment (dennisschagt) [inline src/htmlrenderer.cpp:688]: It looks like there is some kind of rendering issue in Github's diff viewer. The indentation looks quite different from the indentation I'm seeing locally in vim (specifically the indentation of the changed code compared with the switch case indentation)
It looks like only tabs are used for indentation (as expected), so I'm not sure what is going on

image
image

Comment (Minoru) [reply src/htmlrenderer.cpp:688]: Yeah, looks like a GitHub glitch. The file is fine, so this doesn't matter :) Thanks for the heads-up though!

**Comment by Minoru** [comment]
Thanks for the fix! So the issue was caused by closing a `li` tag twice? Could you add a test for that please? :) (It's fine if the test just crashes; Catch2 doesn't let us do any better).

**Comment by dennisschagt** [comment]
> Thanks for the fix! So the issue was caused by closing a `li` tag twice?

It looks like it was caused by a double closed `<ul>` or `<ol>` tag.
That is also how I got this issue to reproduce in a regression test.

> Could you add a test for that please? :) (It's fine if the test just crashes; Catch2 doesn't let us do any better).

Sure, I got a bit distracted, but the requested test is now available.

**Review by Minoru — APPROVED** [review]
Thanks! CI failure on macOS Monterey is fixed by #1984, so I'm merging this.

**Pull request merged**